src/App.tsx

```diff
@@ -15,47 +16,126 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((goodA, goodB): number => {
+    switch (sortType) {
+      case SortType.ALPABET:
+        return goodA.localeCompare(goodB);
+
+      case SortType.LENGTH:
+        return goodA.length - goodB.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const handleSortByAlpabet = () => {
+    if (sortType !== SortType.ALPABET) {
+      setSortType(SortType.ALPABET);
+    } else {
+      setSortType(SortType.NONE);
+    }
```

Comment:
This is redundant code, you will never have `SortType.NONE`, because you always pass one, specific type on a button click. so you don't need to use condition here (and below the same).

```
const handleSortByAlpabet = () => setSortType(SortType.ALPHABET)
```